ui/src/app/machines/views/MachineList/MachineList.js

```diff
@@ -516,6 +519,23 @@ const MachineList = () => {
     showMAC
   };
 
+  let rows;
+  if (!rendered) {
```

Comment:
Did you try setting default rows to render if `!machinesLoaded`?

Reply:
>  We should be responding to state changes, not whether react has rendered yet if possible.

I agree with that in principle, but if we take this example to a logical extreme, say the browser has to render 1000 table rows on initial render and it takes 3s while the screen appears frozen. Wouldn't that warrant a *little* skirt around best practices for a better user experience?

> If we render the table with some default data, and remove the machinesLoaded guard on l553, that should achieve what we want here I think? 

I don't think so, because once those machines have loaded into state the browser would still render all the machines at once every time you navigate to the machine list. The central issue is the amount of time the browser takes to repaint the screen, which I was hoping to address with the "rendered" check.

I will say that this has only been an issue I've noticed in development, and could totally be a non-issue in production.

Reply:
> I will say that this has only been an issue I've noticed in development, and could totally be a non-issue in production.

I think we should definitely confirm that the issue exists in production first @Caleb-Ellis when you have some time - you should be able to just use the test factories to create a bunch of machines on a local maas.